crates/teloxide/src/dispatching/dispatcher.rs

```diff
@@ -495,6 +495,32 @@ impl<R, Err, Key> Dispatcher<R, Err, Key> {
     }
 }
 
+/// Extension methods for adding `tracing` to [`UpdateHandler`].
+///
+/// This trait is only available if the `tracing` feature is enabled.
+#[cfg(feature = "tracing")]
+pub trait UpdateHandlerTracingExt<E> {
+    /// Returns an `UpdateHandler` with tracing enabled.
+    fn with_tracing(self, f: fn(Arc<Update>) -> tracing::Span) -> Self;
```

Comment:
```suggestion
    fn with_tracing(self, f: impl Fn(DependencyMap) -> tracing::Span) -> Self;
```

Two things here:
1. I think providing `DependencyMap` is better, as it is more flexible
2. You generally want to use `Fn` trait bound and not a function pointer

Reply:
Or maybe using the `Injectable` trait from `dptree` is even better (esp given the typeck PR to `dptree` that could make use of the types in the signature)

Reply:
I'm not sure where the `Injectable` trait would go, I'm not too familiar with `dptree`. Would you care to explain your reasoning here?

Reply:
I have a working version using `DependencyMap`, although I don't quite like the fact that I have to clone it in order to pass it down. Do you think it could affect performance? I'm not sure how other way it could be done.

```rust
// Removed comments for easier read.
pub trait UpdateHandlerTracingExt<E> {
    fn with_tracing_span<F>(self, f: F) -> Self
    where
        F: Fn(DependencyMap) -> tracing::Span + Send + Sync + 'static;
}

impl<E: 'static> UpdateHandlerTracingExt<E> for UpdateHandler<E> {
    fn with_tracing_span<F>(self, f: F) -> Self
    where
        F: Fn(DependencyMap) -> tracing::Span + Send + Sync + 'static,
    {
        let description = DpHandlerDescription::entry().merge_chain(self.description());

        dptree::from_fn_with_description(description, move |deps: DependencyMap, cont| {
            self.clone().execute(deps.clone(), cont).instrument(f(deps))
        })
    }
}
```